### PR TITLE
[W5.10][T15-A3]Dickson Chan

### PR DESCRIPTION
**Pull request by itsdickson**
Moved the formatting code from TextUi class to a new Formatter class to suit SRP.

### CODE REVIEW

File: src/seedu/addressbook/ui/Formatter.java [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+
+import java.io.PrintStream;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+import static seedu.addressbook.common.Messages.*;
+import static seedu.addressbook.common.Messages.MESSAGE_INIT_FAILED;
+
+public class Formatter {
```
Comment (okkhoy) [inline src/seedu/addressbook/ui/Formatter.java]: Need header comments for the class and its methods (if the methods are doing some non-trivial stuff)

File: src/seedu/addressbook/ui/Formatter.java [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+        for (String m : message) {
+            out.println(LINE_PREFIX + m.replace("\n", LS + LINE_PREFIX));
+        }
+    }
+
+    /**
+     * Shows the result of a command execution to the user. Includes additional formatting to demarcate different
+     * command execution segments.
+     */
+    public void showResultToUser(CommandResult result) {
```
Comment (okkhoy) [inline src/seedu/addressbook/ui/Formatter.java]: I guess you have mistaken the exercise?
The `Formatter` class should do exclusive formatting and `UI` only the interaction with the user.
The current implementation moved stuff from UI to Formatter, and Formatter is doing more than just formatting (both formatting and showing to user), thus violating SRP

**Comment by okkhoy** [comment]
@itsdickson 
Some comments added. Please ack & close the PR after reading comments.

File: src/seedu/addressbook/ui/Formatter.java (hunk 2 above)
Comment (itsdickson) [reply src/seedu/addressbook/ui/Formatter.java]: noted! i'll re-read on the notes. thanks!

**Pull request closed without merging**